projects/igniteui-angular/src/lib/services/transaction/base-transaction.ts

```diff
@@ -29,6 +30,8 @@ export class IgxBaseTransactionService<T extends Transaction, S extends State> i
      */
     public onStateUpdate = new EventEmitter<StateUpdateEvent>();
 
+    constructor(public cloneStrategy: IDataCloneStrategy = new DefaultDataCloneStrategy()){}
```

Comment:
we do not need this in the constructor. This could be added in the `Transaction` as public property defaulting to `DefaultDataCloneStrategy` (as it is now). When changing grid's `batchEditing` set the clone strategy of the new transaction to what is set in the grid.